### PR TITLE
[Draft] ♻️ Decouple non-asset related modules

### PR DESCRIPTION
**Pull request by christeefy**
Currently, the way non-asset related modules work is that the underlying LLM consumes the input Dagster code and generates the same input with additional items (e.g. schedules, `Definitions`, sensors, etc.). That is the intention anyway.

In practice, sometimes the input code isn't generated verbatim, which leads to overall flaky behaviour of this module. Furthermore, when such modules provide feedback to the LLM (using `dspy.Assert` or `dspy.Suggest`), there is a risk that the LLM would change the input code as well. This effectively increases the blast radius of changes when an otherwise small module is included, and also introduces intractable regressions.

This PR is an simple attempt to combat that. The high-level idea is to ensure that non-asset related modules generate code snippets as *separate* files. This guarantees that otherwise good input Dagster code into the module won't be modified unexpectedly. The results are then combined using a (non-LLM) utility function.

### CODE REVIEW

File: scripts/airflow2dagster/model.py [hunk 1]
```diff
@@ -18,28 +16,34 @@ def __init__(self):
...
+        pred = self.add_materialization_result(pred.output)
 
         # NOTE: From experiments, configs should be generated *after* materialization results.
         # Otherwise, the `AddMaterializationModule`'s output
         # fails to retain the correct, original Config code and removes it
-        pred = self.add_config(pred.dagster_code)
+        pred = self.add_config(pred.output)
+
+        # if "retries" in airflow_code.lower():
+        #     pred = self.add_retry_policy(airflow_code, pred.output)
```
Comment (christeefy) [inline scripts/airflow2dagster/model.py]: I'm commenting this out temporarily until #6 is merged. That will make the retry policy module an asset-related one.

**Comment by christeefy** [comment]
This PR implementation of the API up to this point is a little messy. Some modules return the "full", translated Dagster code thus far, whereas other modules like `AddScheduleModule` and `AddDefinitions` return code snippets that need to later be combined with the "full" code in `Model.forward`. I think it is messy because the caller needs to be aware of whether a particular module returns "full" or partial code.

Perhaps I'll update the interface such that each module is expected to return full code, and the code snippet piece is an implementation detail within each module.

In other words, within each module that needs to generate code snippets:
1. Consume the `input_dagster_code` (and optionally `airflow_code`) thus far
2. Generate the code snippet
3. Validate code snippet
4. Concatenate code snippet with `input_dagster_code` and return

That way, the `Model.forward` method becomes more declarative:
```py
class Model(...):
    def forward(...):
        pred = self.module1(...)
        pred = self.module2(...)
        ...
```

File: scripts/airflow2dagster/constants.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+PLACEHOLDER_MODULE_NAME = "dagster_code"
+PLACEHOLDER_MODULE_FILENAME = f"{PLACEHOLDER_MODULE_NAME}.py"
```
Comment (christeefy) [inline scripts/airflow2dagster/constants.py:2]: I'm also good if these constants are on a per-module level instead of being globally defined here.
